lxd/device/nic_routed.go

```diff
@@ -355,6 +357,26 @@ func (d *nicRouted) Start() (*deviceConfig.RunConfig, error) {
 				revert.Add(func() { np.Delete() })
 			}
 		}
+
+		routes := util.SplitNTrimSpace(d.config[fmt.Sprintf("%s.routes", keyPrefix)], ",", -1, true)
+
+		if len(addresses) > 0 {
```

Comment:
This should be changed to check if there are routes set, and then if there are no addresses (in order to use the first one) then return an error that routes cannot be added because there's no address set.